### PR TITLE
Support `ForEachDataField`, `DeepCopy`, `Merge` for ordered maps.

### PR DESCRIPTION
**Pull request by wenovus**
For merge if there is any overlap it simply errors out -- I don't think there is a clear way of merging them if there is a partial overlap.

For overlaps where the second ordered list is a subset of the first, then I can see it as a valid merge -- however I'm opting for simplicity for now since this seems like a more advanced use case.

### CODE REVIEW

File: ygot/struct_validation_map.go [hunk 1]
```diff
@@ -906,6 +912,73 @@ func validateMap(srcField, dstField reflect.Value) (*mapType, error) {
 	return &mapType{key: st.Key(), value: st.Elem()}, nil
 }
 
+// copyOrderedMap copies srcField into dstField. Both srcField and dstField are
```
Comment (DanG100) [inline ygot/struct_validation_map.go:915]: nit: `srcField` isn't a parameter so this comment is a little hard to follow

**Review by DanG100 — APPROVED** [review]

File: ygot/struct_validation_map.go (hunk 1 above)
Comment (wenovus) [reply ygot/struct_validation_map.go]: Done.
